### PR TITLE
Adding SMORMS3 and SSMORMS3 as Keras optimizers

### PR DESCRIPTION
**Pull request by GregGovit**
I've added two very good optimizers:
- SMORMS3 based on "RMSprop loses to SMORMS3 - Beware the Epsilon!" http://sifter.org/~simon/journal/20150420.html
- SSMORMS3 which stands for Stabilized SMORMS3 and is a personal modification in which the estimation of the gradient's noise is smoothed leading to better convergence

### CODE REVIEW

File: pyannote/audio/optimizers.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# AUTHORS
+# Gregory GELLY - gregory.gelly@limsi.fr
+
+import os.path
```
Comment (hbredin) [inline pyannote/audio/optimizers.py]: Looks like we do not need this ```import```.

File: pyannote/audio/optimizers.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+from keras.optimizers import Optimizer 
+import keras.backend as K
+
+class SMORMS3(Optimizer):
+    '''SMORMS3 optimizer.
+    Default parameters follow those provided in the blog.
+    # Arguments
+        lr: float >= 0. Learning rate.
+        epsilon: float >= 0. Fuzz factor.
+    # References
```
Comment (hbredin) [inline pyannote/audio/optimizers.py]: Shouldn't we also document ```decay``` parameter?

File: pyannote/audio/optimizers.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+class SSMORMS3(Optimizer):
+    '''SSMORMS3 optimizer for Stabilized SMORMS3.
+    Slight modification of SMORMS3 that stabilizes its behavior
+    # Arguments
+        epsilon: float >= 0. Fuzz factor.
+    # References
+        - [RMSprop loses to SMORMS3 - Beware the Epsilon!](http://sifter.org/~simon/journal/20150420.html)
+    '''
+    def __init__(self, epsilon=1e-8, **kwargs):
+        super(SSMORMS3, self).__init__(**kwargs)
```
Comment (hbredin) [inline pyannote/audio/optimizers.py]: So you managed to remove two out of three hyper-parameters? 
And it outperforms the original version? 👍
